src/Streamly/Internal/Data/SVar.hs

```diff
@@ -162,6 +162,8 @@ newtype Count = Count Int64
              , Ord
              )
 
+type StreamWR t m a = Either (RunInIO m, t m a) (t m a)
```

Comment:
Instead of using an `Either` type we can just use the tuple `(RunInIO m, t m a)`, for the `Right` case of the Either the RunInIO state will be the initial state i.e. `svarMrun`.

Reply:
Yeah, that was the initial plan, that way we can remove `svarMrun` from the `SVar` as well.